### PR TITLE
Adds standing orders model and dsl

### PR DESCRIPTION
**Pull request by yuliansavov**
This PR adds the necessary parts for handling standing orders

### CODE REVIEW

File: spec/features/dsl/standing_orders_spec.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  before do
+    client.config.environment = FidorApi::Environment::Future.new
+  end
+
+  describe '#standing_order' do
+    before do
+      stub_fetch_request(
+        endpoint:      %r{/standing_orders/92bf870d-d914-4757-8691-7f8092a77e0e},
+        response_body: { id: '92bf870d-d914-4757-8691-7f8092a77e0e' }
```
Comment (amirmujkic) [inline spec/features/dsl/standing_orders_spec.rb]: Please extract  '92bf870d-d914-4757-8691-7f8092a77e0e' into a let block.

Maybe you can also extract the `subject` 'Hello World' and the `uuid` on spec level?

File: spec/lib/fidor_api/model/standing_order_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            'routing_info' => { 'account_number' => account_number }
+          }
+          expect(standing_order.routing_type).to eq routing_type
+          expect(standing_order.account_number).to eq account_number
+        end
+      end
+
+      describe 'virtual attributes for schedule' do
+        it 'sets the attributes in the schedule hash' do
+          standing_order.rhythm = 'weekly'
```
Comment (amirmujkic) [inline spec/lib/fidor_api/model/standing_order_spec.rb]: Please extract the `rythm`, `runtime_day_of_month`, `runtime_day_of_week`, `ultimate_run` into something like a params hash.

File: lib/fidor_api/client/dsl.rb [hunk 3]
```diff
@@ -9,6 +9,7 @@ module DSL
       autoload :Preauths,           'fidor_api/client/dsl/preauths'
       autoload :Transactions,       'fidor_api/client/dsl/transactions'
       autoload :Transfers,          'fidor_api/client/dsl/transfers'
+      autoload :StandingOrders,     'fidor_api/client/dsl/standing_orders'
```
Comment (ghost) [inline lib/fidor_api/client/dsl.rb]: 💅 Nit: Can you keep the alphabetical ordering please?

File: lib/fidor_api/client/dsl.rb [hunk 4]
```diff
@@ -19,6 +20,7 @@ def self.included(klass)
         klass.include Preauths
         klass.include Transactions
         klass.include Transfers
+        klass.include StandingOrders
```
Comment (ghost) [inline lib/fidor_api/client/dsl.rb]: 💅 Nit: Can you keep the alphabetical ordering please?

File: lib/fidor_api/model/standing_order.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
+require 'fidor_api/model/routing_type_helper'
+
+module FidorApi
+  module Model
+    class StandingOrder < Model::Base
+      include RoutingTypeHelper
+
+      SUPPORTED_ROUTING_TYPES = {
+        'FOS_P2P_ACCOUNT_NUMBER' => %w[account_number]
```
Comment (ghost) [inline lib/fidor_api/model/standing_order.rb]: Shouldn't we add the same possible routing-types as for the transfers?
If so: I'd move the mapping hash into the helper to the helper so we have to only maintain it at one place.

File: lib/fidor_api/model/standing_order.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+          @schedule[attribute]
+        end
+
+        define_method "#{attribute}=" do |value|
+          @schedule ||= {}
+          @schedule[attribute] = value
+        end
+      end
+
+      def parse_errors(body)
```
Comment (ghost) [inline lib/fidor_api/model/standing_order.rb]: This one was copied from the transfers.
Maybe we can move it to a shared location?

**Comment by amirmujkic** [comment]
@yuliansavov Please rebase.

File: lib/fidor_api/model/standing_order.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      attribute :external_uid, :string
+      attribute :amount,       :integer
+      attribute :currency,     :string
+      attribute :subject,      :string
+      attribute :beneficiary,  :json
+      attribute :schedule,     :json
+
+      attribute_decimal_methods :amount
+
+      def beneficiary=(value)
```
Comment (ghost) [inline lib/fidor_api/model/standing_order.rb:19]: Is it possible to move this to the `BeneficiaryHelper`?

File: lib/fidor_api/model/standing_order.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        write_attribute(:beneficiary, value)
+        define_methods_for(SUPPORTED_ROUTING_TYPES[beneficiary['routing_type']])
+      end
+
+      def routing_type
```
Comment (ghost) [inline lib/fidor_api/model/standing_order.rb:24]: Same

File: lib/fidor_api/model/standing_order.rb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        @beneficiary ||= {}
+        @beneficiary.dig('routing_type')
+      end
+
+      def routing_type=(type)
```
Comment (ghost) [inline lib/fidor_api/model/standing_order.rb:29]: Same

File: lib/fidor_api/model/standing_order.rb (hunk 7 above)
Comment (yuliansavov) [reply lib/fidor_api/model/standing_order.rb:19]: Directly, no. I believe it's the 'attribute :beneficiary,  :json' part that does not agree with this(the result is that define_methods_for is not executed). I tried moving the 'beneficiary' and 'routing_type' into a class to be extended by the standing_order, but then problems happened with the 'defined_methods_for' in the helper. At the end I thought that it's best not to risk breaking the transfers and left it like it's now. I can look more into it, if you like.

Comment (ghost) [reply lib/fidor_api/model/standing_order.rb:19]: Ok. Let's keep it as it is for now.
I plan to rework parts of this in near future.

**Review by ghost — APPROVED** [review]

**Pull request merged**